train_text_to_image.py

```diff
@@ -640,10 +665,70 @@ def main():
         num_warmup_steps=num_warmup_steps_for_scheduler,
         num_training_steps=num_training_steps_for_scheduler,
     )
+    
+    from peft import LoraConfig, get_peft_model
+    def get_lora_model(unet, rank=4, lora_alpha=4, target_modules=None):
+        if target_modules is None:
+            # Default target modules for SD UNet
+            target_modules = [
+                "to_q",
+                "to_k",
+                "to_v",
+                "to_out.0",
+                "conv1",
+                "conv2",
+                "conv_shortcut",
+                "conv3",
+                "conv4",
+            ]
+        
+        config = LoraConfig(
+            r=rank,
+            lora_alpha=lora_alpha,
+            target_modules=target_modules,
+            lora_dropout=0.0,
+            bias="none",
+        )
+        return get_peft_model(unet, config)
+
+    # Only apply LoRA if the flag is set
+    if args.use_lora:
+        logger.info("Using LoRA for training")
+        unet = get_lora_model(
+            unet,
+            rank=args.rank,
+            lora_alpha=args.lora_alpha,
+        )
+        # Only train LoRA parameters
+        params_to_optimize = filter(lambda p: p.requires_grad, unet.parameters())
+    else:
+        logger.info("Training full model (no LoRA)")
+        params_to_optimize = unet.parameters()
+
+    if args.skip_action_conditioning:
+        optimizer = optimizer_cls(
+            unet.parameters(),
+            lr=args.learning_rate,
+            betas=(args.adam_beta1, args.adam_beta2),
+            weight_decay=args.adam_weight_decay,
+            eps=args.adam_epsilon,
+        )
+    else:
+        optimizer = optimizer_cls(
+            [
+                {"params": params_to_optimize},
+                {"params": action_embedding.parameters()},
+            ],
+            lr=args.learning_rate,
+            betas=(args.adam_beta1, args.adam_beta2),
+            weight_decay=args.adam_weight_decay,
+            eps=args.adam_epsilon,
+        )
```

Comment:
This is copied from line 624 -> 639. Update the existing code rather than duplicating code